web/src/components/notification-banner/notification-banner.tsx

```diff
@@ -42,28 +52,27 @@ export default function NotificationBanner({
   const [show, setShow] = useState(false);
 
   const el = useRef(null);
+  const { storageKey = '' } =
+    notification.kind == 'banner' && notification.bannerProps;
 
   useEffect(() => {
     setShow(true);
   }, []);
 
+  function hideBanner(storageKey?: string) {
+    setShow(false);
+    storageKey && localStorage.setItem(storageKey, JSON.stringify(true));
+  }
+
   return (
     <Banner
       ref={el}
-      ctaButtonProps={
-        notification.kind == 'banner'
-          ? {
-              ...notification.actionProps,
-              className: 'cta',
-            }
-          : {}
-      }
-      onClose={() => setShow(false)}
+      bannerProps={notification.kind == 'banner' && notification.bannerProps}
+      onClose={() => hideBanner(storageKey)}
       className="notification-banner"
-      style={{ transform: `translateY(${show ? 0 : -100}%)` }}
+      style={{ display: `${show ? 'flex' : 'none'}` }}
```

Comment:
Could we just not render it at all if `show` is false, rather than returning the component with `display: none`?

```jsx
return show && <Banner …
```